### DISCUSSION TITLE
How to insert links using JS client

### DISCUSSION THREAD

**Discussion by htunnicliff** [geldata/gel#3554] [Q&A]
Hello! I have a question about doing inserts using the JavaScript/TypeScript client.

If I want to insert a new record that includes a `link` to a different model, what is the right way to do it using the JS client? 

The docs for EdgeQL give an example of [inserting links](https://www.edgedb.com/docs/edgeql/insert#inserting-links):

```edgeql
insert Villain {
  name := "Doc Ock",
  nemesis := (select Hero filter .name = "Spider-Man")
};
```

However, the docs for [inserts with the EdgeDB JS client](https://www.edgedb.com/docs/clients/01_js/insert) don't include any examples of inserts with links. 

Any ideas?

**Comment by Sikarii** [comment]
Hey, assuming you meant the query builder, you should be able to something along the lines:
```ts
import e from "./edgeql-js"; // generated client
import { createClient } from "edgedb";

const client = createClient();

const query = e.insert(e.Villain, {
  name: "Doc Ock",
  nemesis: e.select(e.Hero, (hero) => ({ filter: e.op(hero.name, "=", "Spider-Man") })),
});

const result = await query.run(client);
```

Hope this helps!
